agent/consul/leader_connect_ca.go

```diff
@@ -858,6 +867,19 @@ func (c *CAManager) UpdateConfiguration(args *structs.CARequest) (reterr error)
 	return nil
 }
 
+// ValidateConfigUpdater is an optional interface that may be implemented
+// by a ca.Provider. If the provider implements this interface, the
+// ValidateConfigurationUpdate will be called when a user attempts to change the
+// CA configuration, and the provider type has not changed from the previous
+// configuration.
+type ValidateConfigUpdater interface {
```

Comment:
aside: I think `ConfigUpdateValidator` is less ambiguous but `ConfigUpdateValidate()` OTOH is less clear

Reply:
Ya, this interface could probably use a better name. It is hard to name things in Consul because we have way too many things in these packages.